### ISSUE TITLE
Change wording in the UI (Bundle, Entity)

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop-contrib/reference#13]
Following different Backdrop core issues, I've read often indications to not use the terms "Entity" and "Bundle" in the UI. In *Reference*, "Bundle" is used in the settings of a reference field:

```
Allowed bundles
Leave unchecked to allow all bundles to be referenced.
```

On the same page, under *Global settings*, you find "Entity":

`Select the entity type to be referenced.`

No idea which alternative terms to use at the moment. @jenlampton ?

**Comment by jenlampton** [comment]
How about we just use more human language?

> Allowed bundles

The "Allowed" label makes it sound like all should be checked to allow all.  How about `Limit to the following` and then we can remove `Leave unchecked to allow all bundles to be referenced.`

> Select the entity type to be referenced.

How about `What would you like to reference?`

**Comment by olafgrabienski** [comment]
I like the direction but am not sure if it's good to put it grammatically as a question. No other idea at the moment though.

**Comment by mikemccaffrey** [comment]
Alright, instead of "Select the entity type to be referenced", I changed it to "Select the type of content to be referenced".

And I used the labels for the entity type and bundle that are available in the entity info array to dynamically label the bundle option on the instance form to things like "Limit selection by type of node".

And below the checkbox, I changed the help text to "Leave unchecked to allow all types to be referenced."

Hopefully, this resolves the concerns!
